File: Dopamine.Controls/CombiLabel.vb
Public Class CombiLabel
    Inherits Label

#Region "Dependency Properties"
    Public Shared ReadOnly FontSize2Property As DependencyProperty = DependencyProperty.Register("FontSize2", GetType(Integer), GetType(CombiLabel), New PropertyMetadata(Nothing))
    Public Shared ReadOnly FontWeight2Property As DependencyProperty = DependencyProperty.Register("FontWeight2", GetType(FontWeight), GetType(CombiLabel), New PropertyMetadata(Nothing))
    Public Shared ReadOnly FontStyle2Property As DependencyProperty = DependencyProperty.Register("FontStyle2", GetType(FontStyle), GetType(CombiLabel), New PropertyMetadata(Nothing))
    Public Shared ReadOnly Content2Property As DependencyProperty = DependencyProperty.Register("Content2", GetType(String), GetType(CombiLabel), New PropertyMetadata(Nothing))
    Public Shared ReadOnly Foreground2Property As DependencyProperty = DependencyProperty.Register("Foreground2", GetType(Brush), GetType(CombiLabel), New PropertyMetadata(Nothing))
#End Region

#Region "Properties"
    Public Property FontSize2 As Integer
        Get
            Return CInt(GetValue(FontSize2Property))
        End Get

        Set(ByVal value As Integer)
            SetValue(FontSize2Property, value)
        End Set
    End Property

    Public Property FontWeight2 As FontWeight
        Get
            Return CType(GetValue(FontWeight2Property), FontWeight)
        End Get

        Set(ByVal value As FontWeight)
            SetValue(FontWeight2Property, value)
        End Set
    End Property

    Public Property FontStyle2 As FontStyle
        Get
            Return CType(GetValue(FontStyle2Property), FontStyle)
        End Get

        Set(ByVal value As FontStyle)
            SetValue(FontStyle2Property, value)
        End Set
    End Property

    Public Property Content2 As String
        Get
            Return CStr(GetValue(Content2Property))
        End Get

        Set(ByVal value As String)
            SetValue(Content2Property, value)
        End Set
    End Property

    Public Property Foreground2 As Brush
        Get
            Return CType(GetValue(Foreground2Property), Brush)
        End Get

        Set(ByVal value As Brush)
            SetValue(Foreground2Property, value)
        End Set
    End Property
#End Region

#Region "Construction"
    Shared Sub New()
        'This OverrideMetadata call tells the system that this element wants to provide a style that is different than its base class.
        'This style is defined in themes\generic.xaml
        DefaultStyleKeyProperty.OverrideMetadata(GetType(CombiLabel), New FrameworkPropertyMetadata(GetType(CombiLabel)))
    End Sub
#End Region
End Class

File: Dopamine.Common/Controls/CombiLabel.cs
using System;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Media;

namespace Dopamine.Common.Controls
{
    public class CombiLabel : Label
    {
        #region Dependency Properties
        public static readonly DependencyProperty FontSize2Property = DependencyProperty.Register("FontSize2", typeof(int), typeof(CombiLabel), new PropertyMetadata(null));
        public static readonly DependencyProperty FontWeight2Property = DependencyProperty.Register("FontWeight2", typeof(FontWeight), typeof(CombiLabel), new PropertyMetadata(null));
        public static readonly DependencyProperty FontStyle2Property = DependencyProperty.Register("FontStyle2", typeof(FontStyle), typeof(CombiLabel), new PropertyMetadata(null));
        public static readonly DependencyProperty Content2Property = DependencyProperty.Register("Content2", typeof(object), typeof(CombiLabel), new PropertyMetadata(null));
        public static readonly DependencyProperty Foreground2Property = DependencyProperty.Register("Foreground2", typeof(Brush), typeof(CombiLabel), new PropertyMetadata(null));
        #endregion

        #region Properties
        public int FontSize2
        {
            get { return Convert.ToInt32(GetValue(FontSize2Property)); }

            set { SetValue(FontSize2Property, value); }
        }

        public FontWeight FontWeight2
        {
            get { return (FontWeight)GetValue(FontWeight2Property); }

            set { SetValue(FontWeight2Property, value); }
        }

        public object Content2
        {
            get { return (object)GetValue(Content2Property); }

            set { SetValue(Content2Property, value); }
        }

        public Brush Foreground2
        {
            get { return (Brush)GetValue(Foreground2Property); }

            set { SetValue(Foreground2Property, value); }
        }
        #endregion

        #region Construction
        static CombiLabel()
        {
            //This OverrideMetadata call tells the system that this element wants to provide a style that is different than its base class.
            //This style is defined in themes\generic.xaml
            DefaultStyleKeyProperty.OverrideMetadata(typeof(CombiLabel), new FrameworkPropertyMetadata(typeof(CombiLabel)));
        }
        #endregion
    }
}
